### PR TITLE
Add dissector for SSH version exchange

### PR DESCRIPTION
**Pull request by x-way**

### CODE REVIEW

**Comment by mcr** [comment]
Looks great. Can you rebase?

**Comment by fxlb** [comment]
Please, use new GET_ macros instead of EXTRACT_ ones (See 176e182 and ee68aa3).

File: print-ssh.c [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+static int
+ssh_print_version(netdissect_options *ndo, const u_char *pptr, u_int len)
+{
+	u_int idx = 0;
+	const char *pnp;
+
+	ND_TCHECK_1(pptr+idx);
+	if ( EXTRACT_U_1(pptr+idx) != 'S' )
+		goto trunc;
```
Comment (fxlb) [inline print-ssh.c]: `trunc` means a truncation of the packet. It's not the case here and in line 40, 44, 48.

**Comment by fxlb** [comment]
The new GET_ macros do the bounds checks with a setjmp/longjmp logic.
```
/* part of extract.h */

static inline uint8_t
get_u_1(netdissect_options *ndo, const u_char *p)
{
        if (!ND_TTEST_1(p))
                longjmp(ndo->ndo_truncated, 1);
        return EXTRACT_U_1(p);
}

[...]

#define GET_U_1(p) get_u_1(ndo, (const u_char *)(p))
```
Thus you can remove the ND_TCHECK_1 macros.

**Comment by fxlb** [comment]
For a better git history, I suggest you to squash all the commit in one with `git rebase -i HEAD~8` (fixup (f) keyword for the last 7 commits)

**Comment by fxlb** [comment]
Merged. Thank you.

**Pull request merged**